lib/SILGen/SILGenConvert.cpp

```diff
@@ -775,8 +775,9 @@ ManagedValue SILGenFunction::emitExistentialErasure(
 
     auto upcast =
       B.createInitExistentialMetatype(loc, metatype,
-                                      existentialTL.getLoweredType(),
-                                      conformances);
+                      existentialTL.getLoweredType(),
+                      cast<MetatypeType>(concreteFormalType).getInstanceType(),
```

Comment:
concreteFormalType.getMetatypeInstanceType()

Reply:
That doesn't give back a canonical type, so I'd have to coerce it back. Not sure that's any cleaner.